src/refract/JsonSchema.cc

```diff
@@ -0,0 +1,290 @@
+//
+//  refract/JsonSchema.cc
+//  librefract
+//
+//  Created by Thomas Jandecka on 16/02/2018
+//  Copyright (c) 2018 Apiary Inc. All rights reserved.
+//
+
+#include "JsonSchema.h"
+
+#include <cassert>
+#include <bitset>
+#include "Element.h"
+#include "../ElementData.h"
+#include "Utils.h"
+#include "../utils/log/Trivial.h"
+
+#include "VisitorUtils.h" // GetValue
+
+using namespace refract;
+using namespace schema;
+using namespace drafter::utils;
+using namespace drafter::utils::log;
+
+namespace
+{ // API Elements tools
+    so::String instanciate(const StringElement& e)
+    {
+        assert(!e.empty());
+        return so::String{ e.get().get() };
+    }
+
+    so::Number instanciate(const NumberElement& e)
+    {
+        assert(!e.empty());
+        return so::Number{ e.get().get() };
+    }
+
+    so::Value instanciate(const BooleanElement& e)
+    {
+        assert(!e.empty());
+        return e.get() ? //
+            so::Value{ in_place_type<so::True>{} } :
+            so::Value{ in_place_type<so::False>{} };
+    }
+
+    bool has_fixed_attr(const IElement& e)
```

Comment:
why snake_naming? Full project with few exceptions used cameCase

Reply:
Because I find it way more readable, even though it is different from the rest of the project. Also, it's not a public interface.
Should I change this in the whole commit?

Reply:
my personal preference is too snake_naming, but projects is in camelCase. 
So I mean we should continue with. Yes in whole commit.